### PR TITLE
Use file context manager to close files in a timely manner

### PR DESCRIPTION
**Pull request by exarkun**
The test suite reports ResourceWarning from two places in the test suite.  This fixes those cases.

### CODE REVIEW

File: src/klein/test/test_resource.py [hunk 1]
```diff
@@ -575,9 +575,10 @@ def test_staticRoot(self) -> None:
         app = self.app
 
         request = MockRequest(b"/__init__.py")
-        expected = open(
+        with open(
             os.path.join(os.path.dirname(__file__), "__init__.py"), "rb"
-        ).read()
+        ) as f:
+            expected = .read()
```
Comment (twm) [inline src/klein/test/test_resource.py:581]:
```suggestion
            expected = f.read()
```

**Review by twm — APPROVED** [review]
Thank you for finding these! Feel free to merge once everything's green.

**Comment by adiroiban** [comment]
I don't know why this is failing on CI.

And I can't reproduce it on my local dev env...
but even if I put a broken config link, it still fails on my local dev.

```
$ sphinx-build -b html -aEvW docs/ htmldocs
Running Sphinx v5.1.1
building [mo]: all of 0 po files
building [html]: all source files
updating environment: [new config] 16 added, 0 changed, 0 removed
reading sources... [100%] release                                                                         
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] release                                                                          
generating indices... genindex done
writing additional pages... search done
copying downloadable files... [100%] introduction/codeexamples/template.py                                
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in htmldocs.
(venv) ✔  ~/chevah/klein [quiet-resource-warnings|✔ ] 

10:46 $ git top
63f7178 Adi Roiban 2022-08-28 Use RTD link for API extension.
7f536e5 Jean-Paul Calderone 2022-08-27 Fix obvious refactoring mistake
```

**Comment by adiroiban** [comment]
I  have pushed some changes to this PR...but think is better to remove them.

I will try to help fixing https://github.com/twisted/klein/pull/565

@exarkun let  me know if  you want me to rewrite the history and remove the last 2 commits from this PR

**Comment by exarkun** [comment]
> I have pushed some changes to this PR...but think is better to remove them.
> 
> I will try to help fixing #565
> 
> @exarkun let me know if you want me to rewrite the history and remove the last 2 commits from this PR

Thanks for looking into this @adiroiban .  Since the changes didn't fix the issue, removing them makes sense.  It doesn't make any difference to me whether they're removed with history rewriting or not.  I usually avoid history rewriting for public branches but force pushing `HEAD^^` sounds quite low-effort so if that's your preference please go ahead - it won't cause any issues for me.

**Comment by glyph** [comment]
It's approved and the issues on master that were gumming up the works are now addressed, so hopefully this will cleanly auto-merge after CI has done its thign.

**Pull request merged**